### PR TITLE
Add error messaging to upload

### PR DESCRIPTION
**Pull request by bamader**
## Background
At some point, we had error messaging that informed a user of an incorrectly formatted file upload. That messaging disappeared, so this PR adds it back in while also simplifying the messaging states/cases to always direct the user to upload data only in one of our templates.

## GitHub Issue
https://github.com/ctoec/data-collection/issues/1355

## Validation Plan
- Open and log into app
- Go to file upload page
- [ ] Try to upload a csv or xlsx that has different formatting from the template
  - [ ] Verify error message appears and directs user to the template
  - [ ] Error alert disappears on page refresh or successful file upload
- [ ] Try to upload a file of the completely wrong type (i.e. a .zip, .jpg, etc.)
  - [ ] Verify error message appears and directs user to the template
  - [ ] Error alert disappears on page refresh or successful file upload

## Automated Testing
- [x] All unit tests are passing.
- [ ] All e2e tests are passing.

## Additional Context
The reason for testing on two different file types is that we have two different error cases in the back end depending on the type of error that happens while parsing. Spreadsheet files with improperly formatted columns will trigger an `ApiError` version of a 400, (lines 62-66 in `src/routes/enrollmentReport.ts`), while files that aren't spreadsheets at all will trigger the more general `BadRequestError` type of 400 (lines 67-69). We want this error message to appear for both of these cases, since the ticket's intention in simplifying the messaging was to handle all 400s by having the user use a proper template.

### CODE REVIEW

**Review by JordanGuinn — APPROVED** [review]

File: client/src/containers/Upload/Upload.tsx [hunk 1]
```diff
@@ -108,6 +112,7 @@ const Upload: React.FC<StepContentProps> = ({
   return (
     <div className="grid-container">
       <BackButton location="/" />
+      {alertElements}
```
Comment (vonbearshark) [inline client/src/containers/Upload/Upload.tsx:115]: welp yeap there's my problem...

**Review by vonbearshark — APPROVED** [review]

**Comment by JordanGuinn** [comment]
## Validation Plan 
- [ ] Confirm no error messaging is present on valid file uploads
- [ ] Confirm error messaging is present on upload of CSV with improper number of columns
- [ ]  Confirm error messaging is present on upload of CSV with improperly named columns
- [ ]  Confirm error messaging is present on upload of invalid file type (zip)

**Comment by JordanGuinn** [comment]
@bamader Functionally all looks well!  Just got a few minor thoughts on the copy here:

- Could the alert title be updated to something like `We were unable to upload your file`?  I think the title should **first** be indicative that we weren't actually able to ingest their data, while the alert's internal text should direct them to our templates
- Instead of including links directly to the CSV/Excel templates here, could the `templates provided` text instead be a link to the data templates page?  Just cuz (imo) the presence of those additional file download links in the alert make it a pretty chunky element, on an already content-heavy page

**Comment by bamader** [comment]
@JordanGuinn Yes to both changes, and both are implemented. E2e's previously passed, so this is ready to merge whenever.

**Comment by bamader** [comment]
@JordanGuinn Am I cool to merge this? Wanna get it in before the sprint rituals this afternoon.

**Comment by JordanGuinn** [comment]
Do it!  Sorry, thought I responded earlier then got REAL sidetracked

**Pull request merged**